### PR TITLE
RF: Move AutomagicIO functionality to -deprecated

### PR DESCRIPTION
**Pull request by mih**
Analog to the WebUI transition, all functionality stays as-is, given the
extension is installed. If the functionality is requested , but the
extension is not installed, users are prompted to install it.

Tests are now placed in the extension too.

Closes gh-4742

when merged, move:

- #2062
- #4742

### CODE REVIEW

**Comment by mih** [comment]
#5578 fixes the travis failure that is already in master. The appveyor failure is the mac installation issue. The crippled FS failure is #5300

File: datalad/__main__.py [hunk 1]
```diff
@@ -73,6 +72,21 @@ def main(argv=None):
...
+    except Exception as e:
+        # we could just test for ModuleNotFoundError (which should be
+        # all that would happen with PY3.6+, but be a little more robust
+        # and use the pattern from duecredit
+        if type(e).__name__ not in ('ImportError', 'ModuleNotFoundError'):
+            lgr.error("Failed to import datalad_deprecated.auto "
+                      "due to %s", str(e))
+        sys.stderr.write("{}\n".format(
+            "The DataLad AutomagicIO functionality has been moved to an "
+            "extension package. Please install the Python package "
```
Comment (bpoldrack) [inline datalad/__main__.py:86]:
```suggestion
            "extension package. Please install or update the Python package "
```

**Review by bpoldrack — APPROVED** [review]
LGTM

File: datalad/__main__.py (hunk 1 above)
Comment (mih) [reply datalad/__main__.py:86]: I think this is fine without the change. There is only a single release of -deprecated, and it has all that is needed. If there is a problem, the message issued by `create-sibling` also needs to be adjusted.

**Comment by mih** [comment]
thx @bpoldrack for the review.

File: datalad/__main__.py (hunk 1 above)
Comment (bpoldrack) [reply datalad/__main__.py:86]: Yeah, it's okay. Just thought might be slightly better to make that a "habit" to not require adjustment.

**Pull request merged**